lib/prawn/text/formatted/parser.rb

```diff
@@ -238,6 +236,14 @@ def self.array_from_tokens(tokens)
           end
           array
         end
+
+        def self.escape(text)
+          text.gsub('&', '&amp;').gsub('>', '&gt;').gsub('<', '&lt;')
```

Comment:
`gsub` can take a hash as the second argument.

```ruby
text.gsub(/&|>|</, '&' => '&amp;', '>', => '&gt;', '<' => '&lt;')
```

This whole hash  can live in a constant. Let's call it `MAPPING` for example.

Then we can reduce it further.

```ruby
MAPPING = {
  '&' => '&amp;',
  '>', => '&gt;',
  '<' => '&lt;',
}.freeze

def self.escape(text)
  text.gsub(Regexp.union(MAPPING.keys), MAPPING)
end

def self.unescape(text)
  mapping = MAPPING.inverse
  text.gsub(Regexp.union(mapping.keys), mapping)
end
```